### DISCUSSION TITLE
Save tree state on re-render

### DISCUSSION THREAD

**Discussion by benayat** [dowjones/react-dropdown-tree-select#527] [Q&A]
**Is your feature request related to a problem? Please describe.**
I started using this component with redux-rtk query,and whenever there's a re-fech of the data,the tree goes to the initial state.is there a way to save and reload the state for each node?

**Describe the solution you'd like**
A prop for keeping the tree state,or initial state.

**Describe alternatives you've considered**
I'm thinking about implementing seperate state for each node,however I'm not sure it's the right way.

**Additional context**

**Comment by mrchief** [comment]
Yes, you can do that with [prevent-re-render-on-parent-prop-state-changes](https://github.com/dowjones/react-dropdown-tree-select/blob/develop/docs/HOC.md#prevent-re-render-on-parent-prop-state-changes)
